src/Spatial/Euclidean/PolyLine2D.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// The PolyLine2D class represents a 2D curve in space made up of line segments joined end-to-end, and is 
+    /// stored as a sequential list of 2D points.
+    /// </summary>
+    public class PolyLine2D : IEnumerable<Point2D>
+    {
+        // Internal storage for the points
+        private List<Point2D> _points;
+
+        /// <summary>
+        /// Returns the number of points in the polyline
+        /// </summary>
+        public int Count => this._points.Count;
+
+        /// <summary>
+        /// Returns the length of the polyline as the sum of the length of the individual segments
+        /// </summary>
+        public double Length => this.GetPolyLineLength();
+
+        /// <summary>
+        /// Default constructor, creates an empty PolyLine2D
+        /// </summary>
+        public PolyLine2D() : this(Enumerable.Empty<Point2D>())
+        {
+        }
+
+        /// <summary>
+        /// Constructor which creates a PolyLine2D off of a pre-existing IEnumerable of Point2Ds
+        /// </summary>
+        /// <param name="points"></param>
+        public PolyLine2D(IEnumerable<Point2D> points)
+        {
+            this._points = new List<Point2D>(points);
+        }
+
+        /// <summary>
+        /// Access a point in the polyline by index number
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public Point2D this[int key]
+        {
+            get { return this._points[key]; }
+            set { this._points[key] = value; }
+        }
+
+        /// <summary>
+        /// Computes the length of the polyline by summing the lengths of the individual segments
+        /// </summary>
+        /// <returns></returns>
+        private double GetPolyLineLength()
+        {
+            double length = 0;
+            for (int i = 0; i < this._points.Count - 1; ++i)
+                length += this[i].DistanceTo(this[i + 1]);
+            return length;
+        }
+
+        /// <summary>
+        /// Get the point at a fractional distance along the curve.  For instance, fraction=0.5 will return
+        /// the point halfway along the length of the polyline.
+        /// </summary>
+        /// <param name="fraction">The fractional length at which to compute the point</param>
+        /// <returns></returns>
+        public Point2D GetPointAtFractionAlongCurve(double fraction)
+        {
+            return this.GetPointAtLengthFromStart(fraction * this.Length);
```

Comment:
Fraction must be 0 < fraction < 1 here right? Check and throw argument exception feels right if so.